File: src/main/kotlin/io/cronenbergworld/idea/plugin/symbol/SymbolTemplateContextType.kt
package io.cronenbergworld.idea.plugin.symbol

import com.intellij.codeInsight.template.TemplateActionContext
import com.intellij.codeInsight.template.TemplateContextType
import org.jetbrains.annotations.Nls
import org.jetbrains.annotations.NotNull

abstract class SymbolTemplateContextType(id: @NotNull String, presentableName: @Nls @NotNull String)
    : TemplateContextType(id, presentableName) {

    override fun isInContext(templateActionContext: TemplateActionContext): Boolean {
        return templateActionContext.file.isWritable
    }
}

File: src/main/scala/io/cronenbergworld/idea/plugin/symbol/SymbolTemplateContextType.scala
package io.cronenbergworld.idea.plugin.symbol

import com.intellij.codeInsight.template._
import org.jetbrains.annotations.Nls

abstract class SymbolTemplateContextType(id: String, @Nls presentableName: String)
    extends TemplateContextType(id, presentableName) {
  override def isInContext(templateActionContext: TemplateActionContext): Boolean =
    templateActionContext.getFile.isWritable
}
